force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +131,17 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        Integer vowelCount = 0;
+        for (Integer counter=0; counter < str.length(); counter++) {
+            switch on str.substring(counter, counter+1) {
+                when 'a', 'e', 'i', 'o', 'u' {
+                    vowelCount += 1;
+                }
+            }
```

Comment:
This is really clever and I am not sure I ever would have thought of using a switch statement.  I think the easiest way to check for membership is to a Set or Map. 

Here is my solution to this problem. 

```
  public static Integer countVowels(String str) {
        // Initialize the result Integer
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        for (String character : str.split('')) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```